src/coreclr/scripts/superpmi_asmdiffs_setup.py

```diff
@@ -167,18 +171,22 @@ def main(main_args):
         os.makedirs(base_jit_directory)
 
     print("Fetching history of `main` branch so we can find the baseline JIT")
-    run_command(["git", "fetch", "origin", "main"], source_directory, _exit_on_fail=True)
+    run_command(["git", "fetch", "--depth=500", "origin", "main"], source_directory, _exit_on_fail=True)
```

Comment:
why this change?

Reply:
This fixes the problem we were seeing not finding a JIT, e.g., https://dev.azure.com/dnceng/public/_build/results?buildId=1475182&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=bf6cf4cf-6432-59cf-d384-6b3bcf32ede2.

The pipelines by default only fetch with `depth=20`. If there aren't any JIT changes in those 20, we wouldn't find a baseline JIT. So, fetching a lot more makes it more likely to find a baseline JIT change.

Reply:
E.g., you can see the initial `fetch` here:

https://dev.azure.com/dnceng/public/_build/results?buildId=1475182&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=01b23941-a81d-5e8f-0e4b-5c0fce1bb894

Reply:
My impression was that `git fetch origin main` (without `--depth` argument) would fetch the entire history. Doesn't how it work?

Reply:
Well, I'm no expert, but it appears the answer is "no". It appears that the first fetch possibly creates a "shallow" repo and you either need to expand that by fetching with a larger depth, or use `fetch --unshallow` to expand it. E.g., from the `fetch` docs:

```
--depth=<depth>
Limit fetching to the specified number of commits from the tip of each remote branch history. If fetching to a shallow repository created by git clone with --depth=<depth> option (see git-clone(1)), deepen or shorten the history to the specified number of commits. Tags for the deepened commits are not fetched.
```